prow/plank/plank.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plank
+
+import (
+	"fmt"
+
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/line"
+)
+
+type kubeClient interface {
+	ListJobs(map[string]string) ([]kube.Job, error)
+	ListProwJobs(map[string]string) ([]kube.ProwJob, error)
+	PatchProwJob(string, kube.ProwJob) (kube.ProwJob, error)
+	CreateJob(kube.Job) (kube.Job, error)
+}
+
+type Controller struct {
+	kc kubeClient
+}
+
+func NewController(kc *kube.Client) *Controller {
+	return &Controller{
+		kc: kc,
+	}
+}
+
+func (c *Controller) Sync() error {
+	pjs, err := c.kc.ListProwJobs(nil)
+	if err != nil {
+		return fmt.Errorf("error listing prow jobs: %v", err)
+	}
+	js, err := c.kc.ListJobs(nil)
+	if err != nil {
+		return fmt.Errorf("error listing jobs: %v", err)
+	}
+	jm := map[string]*kube.Job{}
+	for _, j := range js {
+		jm[j.Metadata.Name] = &j
+	}
+	var errs []error
+	for _, pj := range pjs {
```

Comment:
emmmm isn't you are editing pj within the loop which is not gonna work, since it's not a pointer? Use index?